lib/SPIRV/SPIRVReader.cpp

```diff
@@ -666,117 +666,106 @@ void SPIRVToLLVM::setLLVMLoopMetadata(const LoopInstType *LM,
     assert(NumParam <= LoopControlParameters.size() &&
            "Missing loop control parameter!");
   }
-  if (LC & LoopControlExtendedControlsMask) {
-    while (NumParam < LoopControlParameters.size()) {
-      switch (LoopControlParameters[NumParam]) {
-      case InitiationIntervalINTEL: {
-        // To generate a correct integer part of metadata we skip a parameter
-        // that encodes name of the metadata and take the next one
-        Metadata.push_back(llvm::MDNode::get(
-            *Context,
-            getMetadataFromNameAndParameter(
-                "llvm.loop.ii.count", LoopControlParameters[++NumParam])));
-        break;
-      }
-      case MaxConcurrencyINTEL: {
-        Metadata.push_back(llvm::MDNode::get(
-            *Context, getMetadataFromNameAndParameter(
-                          "llvm.loop.max_concurrency.count",
-                          LoopControlParameters[++NumParam])));
-        break;
+  if (LC & InitiationIntervalINTEL) {
+    // To generate a correct integer part of metadata we skip a parameter
+    // that encodes name of the metadata and take the next one
```

Comment:
Is it still true? You have changed access pattern around `NumParam` from pre-increment to post-increment

Reply:
Thanks, removed the comment